### ISSUE TITLE
Cannot open entity page for generated entities including a field with type enum

### ISSUE THREAD

**Issue by andidev** [jhipster/generator-jhipster#1776]
Here is my entity.json
```
{
    "relationships": [],
    "fields": [
        {
            "fieldId": 1,
            "fieldName": "asdasd",
            "fieldType": "MyEnum4",
            "fieldIsEnum": true,
            "fieldValues": "MYVALUE",
            "fieldNameCapitalized": "Asdasd",
            "fieldNameUnderscored": "asdasd",
            "fieldInJavaBeanMethod": "Asdasd",
            "fieldValidate": false
        }
    ],
    "fieldsContainOwnerManyToMany": false,
    "fieldsContainOneToOne": false,
    "fieldsContainOwnerOneToOne": false,
    "fieldsContainOneToMany": false,
    "fieldsContainLocalDate": false,
    "fieldsContainCustomTime": false,
    "fieldsContainBigDecimal": false,
    "fieldsContainDateTime": false,
    "fieldsContainDate": false,
    "fieldsContainBlob": false,
    "changelogDate": "20150728114841",
    "dto": "no",
    "pagination": "no",
    "validation": false
}
```

I'm using it with a jhipster application with default options. It seems that it is possible to open the entity main page at all. Not sure why since I get no error message.

**Comment by jdubois** [comment]
Do you still have the issue? Does anybody also have this ?
Can you check if you have an error in your browser console ?
I'm closing this as it works for me, and looking at your JSON file you are on an older version

**Comment by ycbcy** [comment]
Hi,

I still have this issue with a freshly and newly created entity (based on 2.19.0 and with wizard).

The field in question is : 

        {
            "fieldId": 3,
            "fieldName": "quality",
            "fieldType": "FormatQuality",
            "fieldValues": "retina,hd,sd,tablet,phone"
        }

There is no message on console because it failed during the init of the stateProvider of the entity using the enum.

by example : 
/[projectName]/src/main/webapp/scripts/app/entities/[entityName]/[entityName].js

I noticed that the line 

$translatePartialLoader.addPart('');

is added and the getAll and detail route state provider.

When commented the entity main page is now well processed.

Hope it helps and my explanation are clear enough.

See Ya

Tigrou

**Comment by jdubois** [comment]
Oh, that might be linked to the new optional i18n option!
I'm reopening this, thank you!

**Comment by ycbcy** [comment]
My Pleasure,

I spent my night on this troubleshooting before finding that was enum which blow my work :package: 

Happy to help

**Comment by deepu105** [comment]
@jdubois its not enableTranslation issue, coz the $translatePartialLoader.addPart(''); blocks are inside `if` check this seems more like some i18n file missing to me

@ycbcy what is your `.yo-rec.json` config and entity config?

**Comment by ycbcy** [comment]
Sorry deepu105 I can't find any yo-rec.json file. 

Did you mean yo-rc.json file instead ?

**Comment by deepu105** [comment]
its Ok i got it reproduced

@andidev  @jdubois  problem is in below snippet
```
<% for (var fieldIdx in fields) {
     if (fields[fieldIdx].fieldIsEnum == true) { %>
    $translatePartialLoader.addPart('<%= fields[fieldIdx].enumInstance %>');<% }} %>
```

this is producing `$translatePartialLoader.addPart('');`

**Comment by deepu105** [comment]
there is no `enumInstance` in the json file as well, hi @jabelardo I think you did the PR for this will you be able to take a look at this?

**Comment by andidev** [comment]
`enumInstance` should be there there as it is an in memory variable retreived from fieldType, see https://github.com/jhipster/generator-jhipster/blob/master/entity/index.js#L1164

**Comment by andidev** [comment]
Though it looks like it is not added to the inmemory field object.

**Comment by andidev** [comment]
@ycbcy and @deepu105 
Thanx for the feedback, I was clueless on this one

**Comment by russTbaker** [comment]
I created an entity with an enum and then generated it in my gateway. I noticed that it would not render. So I took out the enum field and then compared the the <entity>.state.js file and found that in two of the states, the generator had added this to resolve:

$translatePartialLoader.addPart('');

When I took that out, the page rendered fine. 
This is happening in version 3.5.0

**Comment by ruddell** [comment]
@russTbaker I just tested with a newly generated Gateway (translation disabled) and did not have the issue you described.  If you can reproduce it with the latest version, please open a new issue and fill out the template with all of the required information.

**Comment by russTbaker** [comment]
Hi @ruddell,

I had translation enabled, with Spanish, English (US) and Romanian.

Do you still want me to fill out a ticket?

Thanks,

Russ Baker

**Comment by ruddell** [comment]
@russTbaker In the future please open a new ticket so we can easily reproduce it, your issue had a very different setup than this one.

I was able to reproduce it and fixed it in #3879, this should be in the next release.  Nice catch and thank you for your feedback!

**Comment by russTbaker** [comment]
@ruddell, sorry about that, I will definitely open a ticket and thanks so much for fixing that!

-Russ Baker
